AFQ/utils/bin.py

```diff
@@ -68,6 +74,15 @@ def get_default_args(func):
     }
 
 
+def toml_to_val(t):
+    if isinstance(t, str) and len(t) < 1:
+        return None
+    elif isinstance(t, str) and t[0] == '{':
```

Comment:
Oh - I see: this means that our config files are not valid TOML, but rather an extension of TOML that also includes dictionaries. For example, my text editor will flag this line when using a TOML linter. Could we use the valid TOML approach to dicts instead? For example, the "servers" block from the following example in the docs for pytoml (https://pypi.org/project/toml/): 

```
>>> import toml
>>> toml_string = """
... # This is a TOML document.
...
... title = "TOML Example"
...
... [owner]
... name = "Tom Preston-Werner"
... dob = 1979-05-27T07:32:00-08:00 # First class dates
...
... [database]
... server = "192.168.1.1"
... ports = [ 8001, 8001, 8002 ]
... connection_max = 5000
... enabled = true
...
... [servers]
...
...   # Indentation (tabs and/or spaces) is allowed but not required
...   [servers.alpha]
...   ip = "10.0.0.1"
...   dc = "eqdc10"
...
...   [servers.beta]
...   ip = "10.0.0.2"
...   dc = "eqdc10"
...
... [clients]
... data = [ ["gamma", "delta"], [1, 2] ]
...
... # Line breaks are OK when inside arrays
... hosts = [
...   "alpha",
...   "omega"
... ]
... """
>>> parsed_toml = toml.loads(toml_string)
```

Reply:
I am thinking about hot this would work in our current implementation. Here is what registration looks like now: 
```
[REGISTRATION]

b0_threshold = 50

min_bval = ''

max_bval = ''

reg_template = 'mni_T1'

mask_template = true

wm_criterion = "{'lb': {'dti_fa': 0.2}, 'ub': {'dti_md': 0.002}}"
```
And here is maybe how we should do it?
```
[REGISTRATION]

b0_threshold = 50

min_bval = ''

max_bval = ''

reg_template = 'mni_T1'

mask_template = true

  [REGISTRATION.wm_criterion]
      [REGISTRATION.wm_criterion.lb]
        dti_da = 0.2
     [REGISTRATION.wm_criterion.ub]
        dti_md = 0.002
```
This sort of breaks with how we are currently using toml.

Reply:
Also, I think you may be looking at the default TOML file before this PR, which was broken? For me, using this PR, my TOML editor does not flag it as an invalid line. Basically the dictionary is stored as a string:
`"{'lb': {'dti_fa': 0.2}, 'ub': {'dti_md': 0.002}}"`

Reply:
Yes. I think this is what we'd like this to look like. That's both valid TOML and also results in the dicts you'd like to ultimately have.

Reply:
In this case, we could add a comment stating that None is '' and dictionaries should be inside of quotes.

Reply:
Ah yes: that is true. Let me think.

Reply:
Hmm. I think that we can use this as-is for now, but in the long run I think that this will be more maintainable if we use the toml library itself to convert dict-to-toml and toml-to-dict.